### PR TITLE
Additional channel/chat lifecycle events

### LINKED ISSUE #4079: Additional channel/chat lifecycle events (c#)
See [parent](https://github.com/microsoft/botframework-sdk/issues/5892)

### PR DESCRIPTION
**Pull request by denscollo**
Fixes #4079

## Description
- Add the following events to the [TeamsActivityHandler](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs) class.
  - teamArchived
  - teamDeleted
  - teamHardDeleted
  - teamRestored
  - teamUnarchived

**Note:** These changes can't be tested using MS-Teams until the implementation to use these events is done.
## Specific Changes
- Updated [TeamsActivityHandler](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs) class to add.
  - _OnTeamsTeamArchivedAsync_ method.
  - _OnTeamsTeamDeletedAsync_ method.
  - _OnTeamsTeamHardDeletedAsync_ method.
  - _OnTeamsTeamRestoredAsync_ method.
  - _OnTeamsTeamUnarchivedAsync_ method.
- Updated switch cases to add the events ` teamArchived` , ` teamDeleted` , ` teamHardDeleted` , ` teamRestored`  and ` teamUnarchived` , calling the previous methods corresponding to each case.
- Updated [TeamsActivityHandlerTests](https://github.com/microsoft/botbuilder-dotnet/blob/master/tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs) class to add the unit tests for the previous methods.

## Testing
You can see the added tests running in the following image.
image

### CODE REVIEW

**Comment by cleemullins** [comment]
Please fix merge conflicts.

**Comment by gabog** [comment]
Marking as blocked for now until I hear back from Teams.
@denscollo, can you fix the merge conflicts in the meantime so the PR is ready when it gets unblocked?

**Comment by denscollo** [comment]
@gabog Sure, we'll fix the conflicts in a bit. Thanks!

File: tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs [hunk 1]
```diff
@@ -314,7 +314,7 @@ public async Task TestConversationUpdateTeamsChannelRestored()
             // Arrange
             var activity = new Activity
             {
-                Type = ActivityTypes.ConversationUpdate,
+                Type = ActivityTypes.ConversationUpdate,
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs:317]: nit: blank space

**Pull request merged**